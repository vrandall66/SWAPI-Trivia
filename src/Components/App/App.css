
.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}

.WelcomeFormModal {
  align-items: center;
  background-image: url("../../images/ChooseWisely.png");
  background-repeat: no-repeat;
  background-position: left 50% bottom 50%;
  background-size: 50%;
  display: flex;
  justify-content: space-around;
  height: 100vh;
  width: 100%;
}

.WelcomeFormOverlay {
  align-items: center;
  background-color: rgba(128, 128, 128, 0.5);
  color: #ffe81f;
  display: flex;
  font-size: 2em;
  justify-content: space-around;
  margin: 0;
  width: 100vw;
}