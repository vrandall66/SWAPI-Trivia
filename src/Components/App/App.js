import React from "react";
import { getAllMovies } from "../../ApiCalls/apiCalls";
import { Route, Link } from "react-router-dom";
import Home from "../Home/Home";
import Form from "../Form/Form";
import ReactModal from "react-modal";
import MoviesContainer from "../MoviesContainer/MoviesContainer";
import MovieModal from "../MovieModal/MovieModal";
import CharactersContainer from "../CharactersContainer/CharactersContainer";
import UserProfile from "../UserProfile/UserProfile";
import imperialGif from "../../images/ImperialGif.gif";
import "./App.css";

ReactModal.setAppElement("#root");

class App extends React.Component {
  constructor() {
    super();
    this.state = {
      showFormModal: true,
      showPlanetModal: false,
      showUserMenu: false,
      name: "",
      favQuote: "",
      ranking: "",
      favoriteCharacters: [],
      allMovies: [],
      currentMovie: {}
    };
  }

  componentDidMount = () => {
    getAllMovies()
      .then(moviedata => this.setState({ allMovies: moviedata.results }))
      .catch(err => console.log(err));
  };

  createMovieObjects = () => {
    const planets = this.state.allMovies.map(movie => {
      return {
        title: movie.title,
        id: movie.episode_id,
        releaseDate: movie.release_date
      };
    });
    return this.sortMovies(planets);
  };

  sortMovies = movies => {
    return movies.sort((a, b) => {
      return a.id - b.id;
    });
  };

  handleFormSubmit = ({ name, favQuote, ranking }) => {
    this.setState({ name, favQuote, ranking });
    this.hideFormModal();
    this.createMovieObjects();
  };

  resetMovieState = () => {
    this.setState({ currentMovie: {} });
  };

  hideFormModal = () => {
    this.setState({ showFormModal: false });
  };

  updatePlanetModalState = () => {
    this.setState({ showPlanetModal: !this.state.showPlanetModal });
  };

  updateUserMenuState = () => {
    this.setState({ showUserMenu: !this.state.showUserMenu });
  };

  updateCurrentMovie = movie => {
    this.setState({ currentMovie: movie }, this.updatePlanetModalState());
  };

  userLogoutReset = () => {
    this.setState({
      showFormModal: true,
      showPlanetModal: false,
      showUserMenu: false,
      name: "",
      favQuote: "",
      ranking: "",
      favoriteCharacters: [],
      currentMovie: {}
    });
  };

  render() {
    return (
      <div className="App">
        <UserProfile
          isOpen={this.state.showUserMenu}
          userName={this.state.name}
          userFavQuote={this.state.favQuote}
          userRanking={this.state.ranking}
          userFavCharacters={this.state.favoriteCharacters}
          updateUserMenuState={this.updateUserMenuState}
          userLogoutReset={this.userLogoutReset}
        />
        <ReactModal
          isOpen={this.state.showFormModal}
          onRequestClose={this.handleFormSubmit}
          style={{ overlay: {}, content: {} }}
          contentLabel="Welcome Form"
          className="WelcomeFormModal"
          overlayClassName="WelcomeFormOverlay"
        >
          <Form handleFormSubmit={this.handleFormSubmit} />
        </ReactModal>
        <Route exact path="/" component={Home} />
        <Route
          exact
          path="/movies"
          render={() =>
            this.state.allMovies.length === 0 ? (
              <img
                src={imperialGif}
                alt="Loading GIF"
                className="MoviesContainer-loading"
              />
            ) : (
              <MoviesContainer
                movies={this.state.allMovies}
                updatePlanetModalState={this.updatePlanetModalState}
                updateCurrentMovie={this.updateCurrentMovie}
                className="MoviesContainer"
                resetMovieState={this.resetMovieState}
                reactModal={
                  <ReactModal
                    isOpen={this.state.showPlanetModal}
                    style={{ overlay: {}, content: {} }}
                  >
                    <MovieModal
                      currentMovie={this.state.currentMovie}
                      updatePlanetModalState={this.updatePlanetModalState}
                    />
                  </ReactModal>
                }
              />
            )
          }
        />
        <Route
          extact
          path="/movies/:id"
          render={({ match }) => {
            const id = parseInt(match.params.id);
            const episode = this.state.allMovies.find(
              episode => episode.episode_id === id
            );
            return (
              <>
                <CharactersContainer episode={episode} />
              </>
            );
          }}
        />
      </div>
    );
  }
}

export default App;

// {
//   this.state.ideas.length === 0 && (
//     <img
//       src="https://www.bluewaysireland.org/Style%20Library/img/loader.gif"
//       alt="Loading GIF"
//     />
//   )
// }
