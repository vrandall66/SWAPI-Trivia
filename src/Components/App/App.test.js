import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("App", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it("should match the wrapper", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should be able to add a favorite character", () => {
    let character = {
      name: "R2-D2",
      species: "Droid",
      homeworld: {
        homeName: "Naboo",
        homePopulation: "4500000000"
      },
      films: [
        "The Empire Strikes Back",
        "Attack of the Clones",
        "The Phantom Menace",
        "Revenge of the Sith",
        "Return of the Jedi",
        "A New Hope",
        "The Force Awakens"
      ],
      characterid: 3,
      favorite: true
    };

    wrapper.instance().addFavoriteCharacter(character);

    expect(wrapper.state()).toEqual({
      allMovies: [],
      currentMovie: {},
      favQuote: "",
      favoriteCharacters: [
        {
          characterid: 3,
          favorite: true,
          films: [
            "The Empire Strikes Back",
            "Attack of the Clones",
            "The Phantom Menace",
            "Revenge of the Sith",
            "Return of the Jedi",
            "A New Hope",
            "The Force Awakens"
          ],
          homeworld: { homeName: "Naboo", homePopulation: "4500000000" },
          name: "R2-D2",
          species: "Droid"
        }
      ],
      name: "",
      ranking: "",
      showFormModal: true,
      showPlanetModal: false,
      showUserMenu: false
    });
  });

  it("should not add favorite character if character.favorite is false", () => {
    let character = {
      name: "R2-D2",
      species: "Droid",
      homeworld: {
        homeName: "Naboo",
        homePopulation: "4500000000"
      },
      films: [
        "The Empire Strikes Back",
        "Attack of the Clones",
        "The Phantom Menace",
        "Revenge of the Sith",
        "Return of the Jedi",
        "A New Hope",
        "The Force Awakens"
      ],
      characterid: 3,
      favorite: false
    };

    wrapper.instance().addFavoriteCharacter(character);

    expect(wrapper.state()).toEqual({
      allMovies: [],
      currentMovie: {},
      favQuote: "",
      favoriteCharacters: [],
      name: "",
      ranking: "",
      showFormModal: true,
      showPlanetModal: false,
      showUserMenu: false
    });
  })

  it("should be able to remove a favorite character from state", () => {
    let favorites = [
      {
      name: "R2-D2",
      species: "Droid",
      homeworld: {
        homeName: "Naboo",
        homePopulation: "4500000000"
      },
      films: [
        "The Empire Strikes Back",
        "Attack of the Clones",
        "The Phantom Menace",
        "Revenge of the Sith",
        "Return of the Jedi",
        "A New Hope",
        "The Force Awakens"
      ],
      characterid: 3,
      favorite: true
    },
    {
      name: "Darth Vader",
      species: "Human",
      homeworld: {
        homeName: "Tatooine",
        homePopulation: 200000
      },
      films: [
        "The Empire Strikes Back",
        "Revenge of the Sith",
        "Return of the Jedi",
        "A New Hope"
      ],
      characterid: 4,
      favorite: true
    }
  ]
    wrapper.instance.setState({
      favoriteCharacters: favorites
    })

    
  })
});
