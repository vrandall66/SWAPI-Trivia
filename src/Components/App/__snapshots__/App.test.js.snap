// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the wrapper 1`] = `
<div
  className="App"
>
  <UserProfile
    isOpen={false}
    updateUserMenuState={[Function]}
    userFavQuote=""
    userLogoutReset={[Function]}
    userName=""
    userRanking=""
  />
  <Route
    exact={true}
    path="/"
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="WelcomeFormModal"
      closeTimeoutMS={0}
      contentLabel="Welcome Form"
      isOpen={true}
      onRequestClose={[Function]}
      overlayClassName="WelcomeFormOverlay"
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {},
          "overlay": Object {},
        }
      }
    >
      <withRouter(Form)
        handleFormSubmit={[Function]}
      />
    </Modal>
  </Route>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="MovieModal"
    closeTimeoutMS={0}
    isOpen={false}
    overlayClassName="MovieModalOverlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {},
        "overlay": Object {
          "backgroundColor": "rgba(255, 255, 255, 0.75)",
          "bottom": 0,
          "left": 0,
          "position": "fixed",
          "right": 0,
          "top": 0,
        },
      }
    }
  >
    <MovieModal
      currentMovie={Object {}}
      updatePlanetModalState={[Function]}
    />
  </Modal>
  <Route
    exact={true}
    path="/movies"
    render={[Function]}
  />
  <Route
    component={[Function]}
    exact={true}
    path="/favorite-characters"
  >
    <FavoriteCharacters
      addFavoriteCharacter={[Function]}
      characterInfo={Array []}
    />
  </Route>
  <Route
    extact={true}
    path="/movies/:id"
    render={[Function]}
  />
</div>
`;
